### PR TITLE
Update documentation for all `@compiled/` packages

### PR DESCRIPTION
**Pull request by dddlr**
Follow-up to https://github.com/atlassian-labs/compiled/pull/1687 and https://github.com/atlassian-labs/compiled/pull/1688

Click this link to see how the website looks: https://deploy-preview-1689--compiled-css-in-js.netlify.app/

Main change:

* **Update documentation for all `@compiled/` packages** so that the options listed are actually accurate
  * There's a few options that aren't used anymore - I've noted them as such in the website. We may fix this in https://github.com/atlassian-labs/compiled/issues/1690 if we get time
  * For options that are used in both `@compiled/babel-plugin`and `@compiled/webpack-loader` / `@compiled/parcel-*`, I've tried my best to de-duplicate them for a single source of truth
  * Other issues discovered while working on this:
    * https://github.com/atlassian-labs/compiled/issues/1691
    * https://github.com/atlassian-labs/compiled/issues/1692

Below are the other changes I've made. I recommend looking at the commit-by-commit:

* Update "Atomic CSS" page to use object styles, not tagged template strings
* Update `@compiled/react` documentation to explicitly deprecate `styled` and `ClassName` APIs, and make `css`/`cssMap`/`keyframes` more prominent
* Update fonts and font sizes to fit more text on a page (and because Roboto belongs in the early 2010s imo)
  * This is a quick hack to make it more readable; I'm not a professional designer, so it's not the most bang-up job. Please be nice 😆 
  * This also substantially helps readability for the other changes in this PR
  * If we have more time, we can consider (1) migrating the docs to a centralised platform or (2) redesigning the website entirely.
* Change the "Installation" page to recommend Parcel over Webpack
* Update title of 'Media queries and other at rules' to more easily fit the sidebar

### CODE REVIEW

File: website/packages/docs/src/pages/pkg-babel-plugin.mdx [hunk 1]
```diff
@@ -41,41 +41,118 @@ Configure in your Babel config.
...
+#### resolver
+
+A custom resolver used to statically evaluate import declarations, specified as either an object or module path. If this is an object, it should contain a `resolveSync` function with the following type signature: `(context: string, request: string) => string`.
+
+- Type: `string | Resolver`
+- Default: `undefined`
+
+#### sortAtRules
+
+Unused.
```
Comment (dddlr) [inline website/packages/docs/src/pages/pkg-babel-plugin.mdx:158]: I reckon dealing with stuff that's unused, deprecated, obsolete etc would be out of scope of this PR - as a website/docs-focussed PR, I want to avoid making changes to the Compiled packages

I've made an issue so that we can properly clean these up in the future https://github.com/atlassian-labs/compiled/issues/1690

File: website/packages/docs/src/pages/pkg-babel-plugin.mdx [hunk 2]
```diff
@@ -41,41 +41,118 @@ Configure in your Babel config.
...
+#### parserBabelPlugins
+
+Babel parser plugins to be used when parsing the source code. Define these to enable extra babel parsers (for example, typescript).
+See the [babel docs](https://babeljs.io/docs/en/plugins/#syntax-plugins) for more context.
+
+Usually, you will set this option through [`@compiled/parcel-config`] or [`@compiled/webpack-loader`](/webpack-loader) -- see there for examples.
+
+#### processXcss
+
+An option we are only using internally. Don't use!
```
Comment (dddlr) [inline website/packages/docs/src/pages/pkg-babel-plugin.mdx:147]: not sure if there's any point documenting internal options

File: website/packages/docs/src/pages/pkg-eslint-plugin.mdx [hunk 3]
```diff
@@ -32,4 +32,4 @@ export default [
 
 ## Rules
 
-See [here](https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin) for a list of rules in `@compiled/eslint-plugin`.
+See [here](https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin) for a non-exhaustive list of rules in `@compiled/eslint-plugin`.
```
Comment (dddlr) [inline website/packages/docs/src/pages/pkg-eslint-plugin.mdx:35]: Tracked through https://github.com/atlassian-labs/compiled/issues/1691

File: website/packages/docs/src/pages/pkg-webpack-loader.mdx [hunk 4]
```diff
@@ -47,67 +47,58 @@ module.exports = {
...
-Defaults to `false`.
+If set, the value of `extensions` will be used in two places:
 
-### bake: boolean
+1. If `resolver` is not set, then `extensions` will be passed to the default resolver, `enhanced-resolver`.
+2. This is also passed to `@compiled/babel-plugin` under the hood.
 
-Enables baking your styles into JavaScript.
+- Type: `string[]`
+- Default: none, but note that `enhanced-resolver` and `@compiled/babel-plugin` have their own different defaults when the `extensions` option is not set.
```
Comment (dddlr) [inline website/packages/docs/src/pages/pkg-webpack-loader.mdx:80]: The fact that there isn't a single default being applied everywhere does admittedly sound like a bug

[`enhanced-resolver`](https://github.com/webpack/enhanced-resolve?tab=readme-ov-file#resolver-options) uses `['js', 'json', 'node']` as a default, while `@compiled/babel-plugin` uses `['.js', '.jsx', '.ts', '.tsx']` as a default - I made an issue to track it here https://github.com/atlassian-labs/compiled/issues/1692

File: website/packages/docs/src/pages/pkg-babel-plugin-strip-runtime.mdx [hunk 5]
```diff
@@ -29,29 +29,26 @@ Configure in your Babel config.
...
-### compiledRequireExclude: boolean
+#### compiledRequireExclude
 
 When set will prevent additional require (one import per rule) in the bundle and add style rules to meta data.
 This could be used when enabling style sheet extraction in a different configuration for SSR.
 
-Defaults to `false`.
-
-### extractStylesToDirectory
+- Type: `boolean`
```
Comment (kylorhall-atlassian) [inline website/packages/docs/src/pages/pkg-babel-plugin-strip-runtime.mdx:37]: Do we want this same "type: boolean, default: false" elsewhere?  I think it's missing on like `extractStylesToDirectory`, but a bit hard to parse.  No real ask, just a comment.

File: website/packages/docs/src/pages/pkg-babel-plugin.mdx (hunk 2 above)
Comment (kylorhall-atlassian) [reply website/packages/docs/src/pages/pkg-babel-plugin.mdx:147]: I think that's fine to be honest, probably helps the internal team as well, eh?

File: website/packages/docs/src/pages/pkg-react.mdx [hunk 6]
```diff
@@ -21,38 +22,58 @@ npm install @compiled/react
...
+
+Use [`cssMap`](/api-cssmap) to define a map consisting of named CSS rules that can be statically typed and useable with other Compiled APIs.
+
+```jsx
+import { cssMap } from '@compiled/react';
+
+const borderStyleMap = cssMap({
+  none: { borderStyle: 'none' },
+  solid: { borderStyle: 'solid' },
+});
```
Comment (kylorhall-atlassian) [inline website/packages/docs/src/pages/pkg-react.mdx:54]: Does this want an example?

Eg.
```suggestion
});

<div css={borderStyleMap[props.appearance]} />
```

File: website/packages/ui/src/components/anchor.tsx [hunk 7]
```diff
@@ -28,6 +28,7 @@ export const Anchor = ({ children }: { children: string | string[] }): JSX.Eleme
   const context = useContext(AnchorContext);
   const ref = useRef<HTMLElement | null>(null);
 
+  console.log(children);
```
Comment (liamqma) [inline website/packages/ui/src/components/anchor.tsx:31]: Is this `console.log` intended to be here?

**Review by liamqma — APPROVED** [review]

File: website/packages/ui/src/components/anchor.tsx (hunk 7 above)
Comment (dddlr) [reply website/packages/ui/src/components/anchor.tsx:31]: oh god 😆 let me remove that

File: website/packages/docs/src/pages/pkg-babel-plugin-strip-runtime.mdx (hunk 5 above)
Comment (dddlr) [reply website/packages/docs/src/pages/pkg-babel-plugin-strip-runtime.mdx:37]: ye sure thing!

File: website/packages/docs/src/pages/pkg-react.mdx (hunk 6 above)
Comment (dddlr) [reply website/packages/docs/src/pages/pkg-react.mdx:54]: done, ty kylor for the suggestion!

**Review by liamqma — APPROVED** [review]
